cli/azd/pkg/output/formatter.go

```diff
@@ -36,3 +39,36 @@ func NewFormatter(format string) (Formatter, error) {
 		return nil, fmt.Errorf("unsupported format %v", format)
 	}
 }
+
+type contextKey string
+
+const (
+	formatterContextKey contextKey = "formatter"
+	writerContextKey    contextKey = "writer"
+)
+
+func WithFormatter(ctx context.Context, formatter Formatter) context.Context {
+	return context.WithValue(ctx, formatterContextKey, formatter)
+}
+
+func GetFormatter(ctx context.Context) Formatter {
+	formatter, ok := ctx.Value(formatterContextKey).(Formatter)
+	if !ok {
+		return &NoneFormatter{}
+	}
+
+	return formatter
+}
+
+func WithWriter(ctx context.Context, writer io.Writer) context.Context {
+	return context.WithValue(ctx, writerContextKey, writer)
+}
+
+func GetWriter(ctx context.Context) io.Writer {
+	writer, ok := ctx.Value(writerContextKey).(io.Writer)
+	if !ok {
+		return colorable.NewColorableStdout()
```

Comment:
Similar code exist in `AskerConsole` implementation. Can we have just one place where colorable writer is instantiated? And can we add NO_COLOR support?

Reply:
fixed.